packages/modules/product/src/migrations/Migration20241125090957.ts

```diff
@@ -0,0 +1,173 @@
+    )
+
+    this.addSql('drop index if exists "IDX_product_image_deleted_at";')
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_image_deleted_at" ON "image" (deleted_at) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_image_url" ON "image" (url) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_tag_value_unique" ON "product_tag" (value) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_type_value_unique" ON "product_type" (value) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'alter table if exists "product" alter column "status" type text using ("status"::text);'
+    )
+    this.addSql(
+      "alter table if exists \"product\" add constraint \"product_status_check\" check (\"status\" in ('draft', 'proposed', 'published', 'rejected'));"
+    )
+    this.addSql(
+      'alter table if exists "product" alter column "status" set default \'draft\';'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_type_id" ON "product" (type_id) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_collection_id" ON "product" (collection_id) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_product_handle_unique" ON "product" (handle) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'alter table if exists "product_option" alter column "product_id" type text using ("product_id"::text);'
+    )
+    this.addSql(
+      'alter table if exists "product_option" alter column "product_id" set not null;'
+    )
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_option_product_id" ON "product_option" (product_id) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_option_product_id_title_unique" ON "product_option" (product_id, title) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'CREATE INDEX IF NOT EXISTS "IDX_product_option_value_option_id" ON "product_option_value" (option_id) WHERE deleted_at IS NULL;'
+    )
+    this.addSql(
+      'CREATE UNIQUE INDEX IF NOT EXISTS "IDX_option_value_option_id_unique" ON "product_option_value" (option_id, value) WHERE deleted_at IS NULL;'
+    )
+
+    this.addSql(
+      'alter table if exists "product_variant" alter column "weight" type integer using ("weight"::integer);'
+    )
+    this.addSql(
+      'alter table if exists "product_variant" alter column "length" type integer using ("length"::integer);'
+    )
+    this.addSql(
+      'alter table if exists "product_variant" alter column "height" type integer using ("height"::integer);'
+    )
+    this.addSql(
+      'alter table if exists "product_variant" alter column "width" type integer using ("width"::integer);'
+    )
```

Comment:
info: they use to be `numeric` now they are `integer` should we think about it as it change the max value you can use?

Reply:
We talked about using big number instead, wouldn't that be more appropriate?

Reply:
oh sorry, I thought we ended up saying we don't do anything :/

Reply:
Yes, we decided to not do anything because `BigNumber` will create a breaking change at the HTTP layer. Correct me if I am wrong 🙂

Reply:
Yeah, that's right. Don't think we should introduce this as part of this refactor. Let's keep it as is.